### PR TITLE
Add a bunch of test in order to increase the coverage.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: src/test/java/org/apache/commons/lang3/ValidateTest.java [hunk 1]
```diff
@@ -774,6 +774,45 @@ void shouldThrowIllegalArgumentExceptionWithGivenMessageForCollectionWithNullEle
...
+            void shouldNotThrowExceptionForValidIndex() {
+                Validate.validState(true, "The Message");
+            }
+
+            @Test
+            void shouldThrowExceptionForTrueExpression() {
+                assertThrows(
+                        IllegalStateException.class,
+                        () ->    Validate.validState(false, "The Message"));
+
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/ValidateTest.java:792]: Too much whitespace IMO.

File: src/test/java/org/apache/commons/lang3/builder/EqualsBuilderTest.java [hunk 2]
```diff
@@ -443,6 +446,16 @@ public void testObjectRecursiveGenericInteger() {
         assertFalse(new EqualsBuilder().setTestRecursive(true).append(o1_b, o2).isEquals());
     }
 
+    @Test
+    public void testObjectsBypassReflectionClasses() {
+
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/builder/EqualsBuilderTest.java:451]: Too much whitespace.

File: src/test/java/org/apache/commons/lang3/ValidateTest.java [hunk 3]
```diff
@@ -774,6 +774,45 @@ void shouldThrowIllegalArgumentExceptionWithGivenMessageForCollectionWithNullEle
...
+        class WithoutMessage {
+
+            @Test
+            void shouldNotThrowExceptionForTrueExpression() {
+                Validate.validState(true);
+            }
+
+            @Test
+            void shouldThrowExceptionForTrueExpression() {
+                assertThrows(
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/ValidateTest.java:807]: You can put stuff like this all on one line IMO.

File: src/test/java/org/apache/commons/lang3/RangeTest.java [hunk 4]
```diff
@@ -103,6 +103,10 @@ public void testBetweenWithCompare() {
         assertTrue(rbstr.contains("i"), "should contain i");
         assertFalse(rbstr.contains("houses"), "should not contain houses");
         assertFalse(rbstr.contains(""), "should not contain ''");
+
+        assertThrows(
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/RangeTest.java:107]: You can put this all on one line IMO.

File: src/test/java/org/apache/commons/lang3/NotImplementedExceptionTest.java [hunk 5]
```diff
@@ -44,6 +45,9 @@ public void testConstructors() {
         assertCorrect("Issue in (Throwable, String)", nie, nested.toString(), nested, code);
         nie = new NotImplementedException(message, nested, code);
         assertCorrect("Issue in (String, Throwable, String)", nie, message, nested, code);
+
+        final NotImplementedException nieNullCode = new NotImplementedException();
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/NotImplementedExceptionTest.java:49]: Do we really need the local variable?

File: src/test/java/org/apache/commons/lang3/ClassUtilsTest.java [hunk 6]
```diff
@@ -410,6 +420,9 @@ class Named {
         }.getClass()));
         assertEquals("org.apache.commons.lang3", ClassUtils.getPackageCanonicalName(Named.class));
         assertEquals("org.apache.commons.lang3", ClassUtils.getPackageCanonicalName(Inner.class));
+        assertEquals(StringUtils.EMPTY, ClassUtils.getPackageCanonicalName((Class<?>) null));
+
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/ClassUtilsTest.java:424]: Remove extra two blank lines.

File: src/test/java/org/apache/commons/lang3/ClassUtilsTest.java [hunk 7]
```diff
@@ -52,6 +52,8 @@
 @SuppressWarnings("boxing") // JUnit4 does not support primitive equality testing apart from long
 public class ClassUtilsTest  {
 
+    private static final String DEFAULT_CANONICAL_NAME = "java.lang.Object";
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/ClassUtilsTest.java:55]: Maybe `DEFAULT_CANONICAL_NAME` -> `OBJECT_CANONICAL_NAME`?

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @arturobernalg,
Thank you for your PR. If you think this PR is still needed, please rebase on master so we can try to get a green build, plus, address my comments.

**Comment by arturobernalg** [comment]
Fix all the remarks @garydgregory 
ty.

File: src/test/java/org/apache/commons/lang3/ArchUtilsTest.java [hunk 8]
```diff
@@ -113,6 +113,10 @@ public void testArchLabels() {
     public void testGetProcessor() {
         assertNotNull(ArchUtils.getProcessor(X86));
         assertNull(ArchUtils.getProcessor("NA"));
+
+        final Processor processor = ArchUtils.getProcessor();
+        assertTrue(processor.isX86());
```
Comment (tisonkun) [inline src/test/java/org/apache/commons/lang3/ArchUtilsTest.java:118]: This test will fail on machines with other arches (e.g. Apple M1 returns `Processor.Type.AARCH_64`). I suggest we remove it.

cc @garydgregory @arturobernalg

**Pull request merged**